### PR TITLE
Add govuk_date_time_field based off govuk_date_field

### PR DESCRIPTION
**Pull request by AlanCornthwaiteKatalyst**

### CODE REVIEW

File: lib/govuk_design_system_formbuilder/elements/datetime.rb [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+
+      private
+
+      def fieldset_options
+        { legend: @legend, caption: @caption, described_by: [error_id, hint_id, supplemental_id],
+          data:   { controller: "datetimes" } }
+      end
+
+      def date
+        tag.div(class: %(#{brand}-date-input)) do
```
Comment (hasarindaKI) [inline lib/govuk_design_system_formbuilder/elements/datetime.rb:48]: `date-time-input` would be preferable.

File: lib/govuk_design_system_formbuilder/elements/datetime.rb [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+          safe_join([day, month, year, hour, minute, meridian, offset])
+        end
+      end
+
+      def omit_day?
+        @omit_day
+      end
```
Comment (hasarindaKI) [inline lib/govuk_design_system_formbuilder/elements/datetime.rb:55]: Not sure we'd want to support this - feels like date-time should always have a date.

**Comment by sobakasu** [comment]
why not add to https://github.com/katalyst/govuk-formbuilder ?

File: lib/govuk_design_system_formbuilder/elements/datetime.rb [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+      def label_classes
+        build_classes(%(label), %(date-input__label)).prefix(brand)
+      end
+    end
+
+    def date
+      tag.div(class: %(#{brand}-date-input)) do
+        safe_join([day, month, year, hour, minute])
+      end
+    end
```
Comment (hasarindaKI) [inline lib/govuk_design_system_formbuilder/elements/datetime.rb:274]: duplicated?

File: lib/govuk_design_system_formbuilder/elements/datetime.rb [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+    #     hint: { text: 'Leave this field blank if you don't know exactly' } do
+    #
+    #       p.govuk-inset-text
+    #         | If you don't fill this in you won't be eligable for a refund
+    #
+    # @example A date input with legend supplied as a proc
+    #  = f.govuk_date_time_field :finishes_on,
+    #    legend: -> { tag.h3('Which category do you belong to?') }
+    def govuk_date_time_field(attribute_name, hint: {}, label: {}, caption: {}, form_group: {}, omit_day: false,
+                              maxlength_enabled: false, **kwargs, &block)
```
Comment (hasarindaKI) [inline lib/govuk_design_system_formbuilder/elements/datetime.rb:334]: Prefer `maxlength_enabled` to be true by default.

File: lib/govuk_design_system_formbuilder/elements/datetime.rb [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+      MULTIPARAMETER_KEY = { day: 3, month: 2, year: 1, hour: 4, minute: 5, offset: 8 }.freeze
+
+      def initialize(builder, object_name, attribute_name, legend:, caption:, hint:, omit_day:, maxlength_enabled:,
+                     form_group:, date_of_birth: false, **kwargs, &block)
+        super(builder, object_name, attribute_name, &block)
+
+        @legend            = legend
+        @caption           = caption
+        @hint              = hint
+        @date_of_birth     = date_of_birth
```
Comment (hasarindaKI) [inline lib/govuk_design_system_formbuilder/elements/datetime.rb:25]: Not sure of the functionality of this - is this needed?

**Review by hasarindaKI — CHANGES_REQUESTED** [review]
Minor comments, nice work 👍

**Comment by sfnelson** [comment]
@AlanCornthwaiteKatalyst this is too complicated to add to Koi without experience using in production-level code.

I noticed specifically:
* copy-paste code from GOVUK (e.g. birthday completion)
* complex code without tests
* complex client-side validation without javascript tests

I don't think this code is ready to add to Koi. Let's discuss options.

**Pull request closed without merging**